pydoctor/epydoc/markup/epytext.py

```diff
@@ -445,8 +478,9 @@ def _add_section(
     # Colorize the heading
     head = _colorize(heading_token, errors, 'heading')
 
-    # Add the section's and heading's DOM elements.
-    sec = Element('section')
+    # Add the section's and heading's DOM elements. 
+    # Add 'ids' attribute just like in docutils.
+    sec = Element('section', ids=[slugify(' '.join(gettext(head)))])
```

Comment:
Special care should be taken to ensure we don't generate twice the same ID for the section header.